h2o-algos/src/main/java/hex/aggregator/AggregatorModel.java

```diff
@@ -111,6 +117,30 @@ public void map(Chunk c, Chunk c2) {
     return res;
   }
 
+  public Frame createMappingOfExemplars(Key destinationKey){
+    final long[] keep = new long[_exemplars.length];
+    for (int i=0;i<keep.length;++i)
+      keep[i]=_exemplars[i].gid;
+
+    Vec exAssignment = _exemplar_assignment_vec_key.get();
+    Arrays.sort(keep);
+    Vec exemplarAssignment = new MRTask() {
+      @Override
+      public void map(Chunk c1, Chunk c2) {
+        for (int i = 0; i < c1._len; i++) {
+          long gid = c1.at8(i);
+          c2.set(i, ArrayUtils.find(keep, gid));
+        }
+      }
+    }.doAll(new Vec[] { exAssignment, exAssignment.makeZero()})._fr.vec(1);
```

Comment:
You don't need to create the zero chunk. It is probably better to use map(Chunk c, NewChunk nc) and use `append` instead of `set` on the `nc`.